hs_core/page_processors.py

```diff
@@ -107,8 +107,8 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
         show_content_files = user.uaccess.can_view_resource(content_model)
     else:
         # if anonymous user getting access to a private resource (since resource is discoverable),
-        # then don't show content files
-        show_content_files = content_model.raccess.public
+        # then don't show content files unless private link sharing is enabled
+        show_content_files = content_model.raccess.public or content_model.raccess.allow_private_sharing
```

Comment:
Wondering whether this allow_private_sharing check should be included in ```user.uaccess.can_view_resource(content_model)```

Reply:
Since `can_view_resource()` checks for an active hydroshare user (https://github.com/hydroshare/hydroshare/blob/develop/hs_access_control/models/user.py#L1425) even for public resources, we can't put the logic to check for `allow_provate_sharing` there.

Reply:
Yeah, makes sense, but is there benefit to update `can_view_resource()` to not check `self.user` if allow_private_sharing is True? I am thinking if we want to support this allow_private_sharing flag in REST API, putting the check here in page processor will not work and it is best to update can_view_resource() to take into account this flag.

Reply:
The REST API endpoints use `authorize()` and the `allow_private_sharing` check is baked into that function.